exporter/otlpexporter/otlp.go

```diff
@@ -97,7 +98,12 @@ func (e *baseExporter) pushTraces(ctx context.Context, td ptrace.Traces) error {
 	}
 	partialSuccess := resp.PartialSuccess()
 	if !(partialSuccess.ErrorMessage() == "" && partialSuccess.RejectedSpans() == 0) {
-		return consumererror.NewPermanent(fmt.Errorf("OTLP partial success: \"%s\" (%d rejected)", resp.PartialSuccess().ErrorMessage(), resp.PartialSuccess().RejectedSpans()))
+		// TODO: These should be counted as "rejected", similar to dropped items.
```

Comment:
If there is an issue tracking this can we include it in the todo?

Reply:
I've updated the title of #9243 to include the additional scope. This PR would address the first half.

The reason this second half is tricky to fix now, is that exporters are constrained by the exporterhelper in terms of metrics they generate. Exporterhelper has no way to distinguish partial success, so without addressing that first, or expanding scope inside this PR, we have the option of counting all-success or all-failed. I believe all-success is the better option until we have a better solution.

I referred to this situation as a "success error", a case where the exporter wants the exporterhelper to see success (i.e., no error) but would like to modify the counts that are applied as it generates metrics. I am not sure how we should approach this from the `exporterhelper` perspective, @dmitryax -- I see in https://github.com/open-telemetry/opentelemetry-collector/pull/9041 there is discussion about adding additional fields to _error responses_, but this is a case calling for additional fields to _success responses_. It would require a new code style if every receiver/processor has to remember that some exporter errors are really successes, meaning to introduce special-case errors with "OK" status to convey partial success.